### ISSUE TITLE
If version number used with the unpublish command, PLEASE warn that it won't be just that version unpublished!

### ISSUE THREAD

**Issue by kshetline** [microsoft/vscode-vsce#533]
I mistakenly thought I could unpublish individual versions of an extension in the Visual Studio Market Place.

How wrong I was. 😢

I typed `vsce unpublish kshetline.ligatures-limited.1.4.1`, hoping to get rid of a copy of my extension that I'd just published, but with some temporary code accidentally left in.

Since the version number was accepted without complaint, I assumed the unpublish command was using the version number I provided.

Now updates for everyone using my extension are completely broken, because everything was wiped out.

The only way for someone to update my extension is to be aware that they need to manually delete the old copy they have, and reinstall the extension from the Market Place. And there's no good way to make the users of my extension know that procedure is necessary.

As far as many users are concerned, they'll simply think updates have stopped being produced. I've published my extension again, but behind the scenes it has a new, unique ID number. As far as Visual Studio Code is concerned, the currently published extension is a totally different extension, one that just happens to have the same name and same author as the one the old version users currently have installed.

I've also lost my download count and my good reviews, which also sucks, but doesn't suck as badly as knowing automatic updates are broken for everyone.

**Comment by joaomoreno** [comment]
> Since the version number was accepted without complaint, I assumed the unpublish command was using the version number I provided.

You should've seen the following message:

```
This will FOREVER delete 'kshetline.ligatures-limited'! Are you sure? [y/N] 
```

> Now updates for everyone using my extension are completely broken, because everything was wiped out.

Maybe @prashantvc, from the Marketplace team, can help you in recovering your lost extension.

**Comment by kshetline** [comment]
Of course, I definitely saw the scary warning, and obviously answered yes, but I still hoped that it applied *only* to the version I'd typed in, or that, because I'd added the version, the command would fail because there is no extension with the full name "kshetline.ligatures-limited.4.1.0".

It seems to me that some part of the vsce code must have actively stripped off the ".4.1.0" part of what I typed. I'm not trying to absolve myself of blame here, but what I'm requesting is that instead of silently stripping off version numbers that users type, the command should either fail, or add to the warning that _all_ versions, not just the specified version, will be lost FOREVER.

Thank you very much for the suggested contact!
